plugins/optimization/src/main/java/org/qbicc/plugin/opt/ea/EscapeAnalysisState.java

```diff
@@ -0,0 +1,88 @@
+package org.qbicc.plugin.opt.ea;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.qbicc.context.AttachmentKey;
+import org.qbicc.context.CompilationContext;
+import org.qbicc.graph.Call;
+import org.qbicc.graph.Executable;
+import org.qbicc.graph.New;
+import org.qbicc.type.definition.element.ExecutableElement;
+
+public final class EscapeAnalysisState {
+    private static final AttachmentKey<EscapeAnalysisState> KEY = new AttachmentKey<>();
+    private final Map<ExecutableElement, List<Call>> callGraph = new ConcurrentHashMap<>();
+    private final Map<ExecutableElement, ConnectionGraph> connectionGraphs = new HashMap<>();
+
+    void trackMethod(ExecutableElement element, ConnectionGraph connectionGraph) {
```

Comment:
Yeah, this is only called once per `element`, since `EscapeAnalysisIntraMethodBuilder` instances are created only 1 per method.

I could add such a check, but I don't currently see a use case where that happens.

Reply:
I like those kinds of checks as runtime enforced documentation.